### PR TITLE
Drafter integration

### PR DESCRIPTION
**Pull request by klokane**
Replace snowcrash by drafter
Use simplified C interface

### CODE REVIEW

**Comment by zdne** [comment]
@klokane tests failing, is this still work in progress?

**Comment by klokane** [comment]
Look like rubocop doesn't like my formating :)
Fixing

**Comment by klokane** [comment]
Looks ready for review

File: lib/redsnow.rb [hunk 1]
```diff
@@ -38,20 +38,15 @@ def self.parse(raw_blueprint, options = 0)
 
     blueprint_options = parse_options(options)
 
-    blueprint = FFI::MemoryPointer.new :pointer
-    sourcemap = FFI::MemoryPointer.new :pointer
     report = FFI::MemoryPointer.new :pointer
 
-    RedSnow::Binding.sc_c_parse(raw_blueprint, blueprint_options, report, blueprint, sourcemap)
+    RedSnow::Binding.drafter_c_parse(raw_blueprint, blueprint_options, report)
```
Comment (pksunkara) [inline lib/redsnow.rb]: Please rename the `report` variable to `parse_result`.

File: lib/redsnow/parseresult.rb [hunk 2]
```diff
@@ -17,51 +19,36 @@ class ParseResult
     # Supported version of Api Blueprint
     SUPPORTED_VERSIONS = ['2.1']
 
-    # @param report_handle [FFI::Pointer]
-    def initialize(report_handle, blueprint_handle, sourcemap_handle)
-      @ast = Blueprint.new(blueprint_handle)
-      @sourcemap = RedSnow::Sourcemap::Blueprint.new(sourcemap_handle)
+    # @param report [ string or nil ]
+    def initialize(report)
```
Comment (pksunkara) [inline lib/redsnow/parseresult.rb]: Similarly, the variable `report` here should be `parse_result`.

File: test/redsnow_binding_test.rb [hunk 3]
```diff
@@ -1,49 +1,23 @@
...
 # RedSnowBindingTest
 class RedSnowBindingTest < Test::Unit::TestCase
   context 'RedSnow Binding' do
     should 'convert API Blueprint to AST' do
       report = FFI::MemoryPointer.new :pointer
-      blueprint = FFI::MemoryPointer.new :pointer
-      sourcemap = FFI::MemoryPointer.new :pointer
 
-      RedSnow::Binding.sc_c_parse("meta: data\nfoo:bar\n#XXXX\ndescription for it", 4, report, blueprint, sourcemap)
+      RedSnow::Binding.drafter_c_parse("meta: data\nfoo:bar\n#XXXX\ndescription for it", 4, report)
```
Comment (pksunkara) [inline test/redsnow_binding_test.rb]: Again, `report` -> `parse_result`. Similarly below.

**Comment by pksunkara** [comment]
Minor variable renames required. Otherwise looks good to me.

**Comment by pksunkara** [comment]
And please don't make the PR to the master branch. Make it to the `shared/attributes` branch.

**Comment by pksunkara** [comment]
Closing this. Please re-open PR to `shared/attributes` branch.

**Pull request closed without merging**